### PR TITLE
Fix #45 and update documentation

### LINKED ISSUE #45: Logger problem: org.slf4j.helpers.SubstituteLogger cannot be cast to ch.qos.logback.classic.Logger
The `protected val logger = LoggerFactory.getLogger(this.getClass).asInstanceOf[Logger]` unsafe cast within `ReactiveRepository` class results in the following runtime error when used with a later version of slf4j (in my case, 1.7.10), and probably in a number of other situations.

> java.lang.ClassCastException: org.slf4j.helpers.SubstituteLogger cannot be cast to ch.qos.logback.classic.Logger

I don't believe the cast itself is necessary as the interface is satisfied before its casted.

See https://travis-ci.org/mefellows/respite/jobs/85068317#L1068 for an example broken build.

### PR DESCRIPTION
**Pull request by mefellows**
* Fixes #45
* Updated documentation as it wasn't clear to me how to find the more recent versions of your lovely package, and also noted a type of 'Responsive' instead of 'Reactive'.

Let me know if you want me to send these through as separate commits / PRs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,12 +18,14 @@ there is a specialized collection called `JSONCollection` that deals naturally w
...
 #### Add simple-reactivemongo
 
-In your project/Build.scala:
+In your project/build.sbt:
 
 ```scala
+resolvers += Resolver.bintrayRepo("hmrc", "releases")
+
 libraryDependencies ++= Seq(
-  "uk.gov.hmrc" %% "simple-reactivemongo" % "[INSERT_VERSION]",
```
Comment (duncancrawford) [inline README.md]: the reason it is "[INSERT_VERSION]" was because we had some ask us where version "x.x.x" was

Comment (mefellows) [inline README.md]: Apologies that looks like a regression, I'll fix that up.

Also LOL.

**Comment by duncancrawford** [comment]
Please fix merge conflicts

**Comment by mefellows** [comment]
Fixed those conflicts, diff looks much nicer now.

**Comment by mefellows** [comment]
Thanks! Could you please update this when you push out the next release to Bintray (or is it automated?).

**Comment by duncancrawford** [comment]
@pacenick could you do me a favour and create a release version for @mefellows?

**Pull request merged**